connector/mqtt/vernemq/README.md

```diff
@@ -0,0 +1,306 @@
+# IoT-Agent MQTT (VerneMQ)
+
+## Overview
+
+An IoT-Agent is an adaptation service between physical devices and the dojot platform. The IoT-Agents are responsible for receiving messages from physical devices (directly or through a gateway) and sending them commands in order to configure. The dojot platform can have multiple IoT-Agents, each one of them being specialized in a specific protocol like in this case MQTT. It is also responsible for ensuring secure communication with devices.
+
+The IoT-Agent MQTT extends [VerneMQ](https://github.com/vernemq/vernemq) with some features and services for dojot case. Basically, devices publishes MQTT messages in specific topics in VerneMQ. These messages are consumed by V2K-bridge service, which adapt them to the dojot's data model and forwards the modified messages to specific topics in Apache Kafka so that they can be consumed by other services. There is a reverse flow, where services publish messages to Apache Kafka in specific topics, these messages are consumed by the service K2V-bridge which adapts the messages and forwards the modified messages to VerneMQ so that they can be consumed by the devices. These data flows are depicted in Fig. 1.
+
+The currently accepted **MQTT protocol versions** are MQTT v3.1 and v3.1.1 respectively.
+
+![image](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/dojot/dojot/epic-100kMqttDevices/connector/mqtt/vernemq/docs/plant_uml/mqtt/diag_mqtt_1)
+
```

Comment:
You can look at the image before merging here
http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/mprevide/dojot/readme_mqtt/connector/mqtt/vernemq/docs/plant_uml/mqtt/diag_mqtt_1